server/__tests__/reviewDataController.test.js

```diff
@@ -0,0 +1,99 @@
+    });
+    server = app.listen(5003, () => console.log('Test server started'));
+});
+
+afterAll(async () => {
+    await server.close();
+    await mongoose.disconnect();
+});
+
+describe('GET /api/reviews/info/:workspaceId', () => {
+    test('should get review data with properties avg and reviewCount', async () => {
+        const res = await request(app).get('/api/reviews/info/1');
+
+        expect(res.status).toBe(200);
+        expect(res.body).toHaveProperty('avg');
+        expect(res.body).toHaveProperty('reviewCount');
+    });
+
+    test('should return statusCode 404 with error message if no data is found', async () => {
+        const res = await request(app).get('/api/reviews/info/10000');
+
+        expect(res.status).toBe(404);
+        expect(res.body).toHaveProperty('message');
+        expect(res.body.message).toBe(noReviewData);
+    });
+});
+
+describe('POST /api/reviews/info', () => {
+    test('should create a new record for ReviewData', async () => {
+        const reviewDataBeforePost = await ReviewData.find();
+        const res = await request(app).post('/api/reviews/info');
+        const reviewDataAfterPost = await ReviewData.find();
+
+        expect(res.status).toBe(201);
+        expect(res.body).toHaveProperty('total');
+        expect(res.body).toHaveProperty('reviewCount');
+        expect(res.body).toHaveProperty('_id');
+        expect(res.body).toHaveProperty('workspaceId');
+        expect(res.body).toHaveProperty('workspaceSlug');
+        expect(res.body).toHaveProperty('reviews');
+        expect(reviewDataAfterPost.length - reviewDataBeforePost.length).toBe(1);
+    });
+});
+
+describe('DELETE /api/reviews/info/:workspaceId', () => {
+    test('should delete ReviewData record', async () => {
+        const existingRecord = await request(app).get('/api/reviews/info/101');
+        expect(existingRecord.body).toHaveProperty('avg');
+        expect(existingRecord.body).toHaveProperty('reviewCount');
+
+        const res = await request(app).delete('/api/reviews/info/101');
+        expect(res.status).toBe(204);
+
+        const deletedRecord = await request(app).get('/api/reviews/info/101');
+        expect(deletedRecord.status).toBe(404);
+        expect(deletedRecord.body.message).toBe(noReviewData);
+    });
+});
+
+describe('PUT /api/reviews/info/:workspaceId', () => {
+    test('should delete a single review in the array of reviews on ReviewData', async () => {
+        const resPost = await request(app).post('/api/reviews/info');
+        const reviewDataItem = await ReviewData.findById(resPost.body._id);
+        const review = {
+            author: 'Bob',
+            date: new Date(),
+            rating: 2,
+            content: 'content',
+            parentId: reviewDataItem._id,
+        };
+
+        reviewDataItem.reviews.push(review);
+        await reviewDataItem.save();
+
+        const updatedReviewDataItem = await ReviewData.find().limit(1).sort({ $natural: -1 });
```

Comment:
I've never seen the $natural parameter within mongoose's sort method. What does this do exactly?

Reply:
It tells mongo to query what I am asking for in reverse order. So here I want the last ReviewData object in the document store.